core/src/array/array_schema.cc

```diff
@@ -2575,6 +2626,48 @@ bool ArraySchema::is_contained_in_tile_slab_row(const T* range) const {
   return true;
 }
 
+void ArraySchema::set_default() {
+  // Clear all members
+  clear();
+
+  // Set array name
+  array_name_ = "";
+  // Set attribute
+  attributes_.emplace_back("a");
+  attribute_num_ = 1;
+  // Set capacity
+  capacity_ = INT_MAX;
+  // Set dimension
+  dimensions_.emplace_back("d");
+  dim_num_ = 1;
+  // Set compression
+  compressor_.emplace_back(Compressor::NO_COMPRESSION);
+  compressor_.emplace_back(Compressor::NO_COMPRESSION);
+  // Set compression level
+  compression_level_.emplace_back(0);
+  compression_level_.emplace_back(0);
+  // Set dense
+  dense_ = 0;
+  // Set number of values per cell
+  cell_val_num_.emplace_back(Configurator::var_num());
+  // Set types
+  types_.emplace_back(Datatype::CHAR);
+  types_.emplace_back(Datatype::INT32);
+  // Set tile extents
+  tile_extents_ = malloc(sizeof(int));
```

Comment:
use `operator new`?  If it is a container type (of fixed size) you can use `std::array`.